brax/jumpy.py

```diff
@@ -30,17 +33,21 @@
 float32 = onp.float32
 int32 = onp.int32
 
+FORCE_JNP = os.getenv("JUMPY_FORCE_JNP", 'False').lower() in ('true', '1', 't')
 
 def _in_jit() -> bool:
   """Returns true if currently inside a jax.jit call."""
+  if FORCE_JNP:
+    return True
   return core.cur_sublevel().level > 0
 
 
 def _which_np(*args):
-  """Returns np or jnp depending on args."""
-  for a in args:
-    if isinstance(a, jnp.ndarray) and not isinstance(a, onp.ndarray):
-      return jnp
+  checker = lambda a: (
+    isinstance(a, (jnp.ndarray, jax.interpreters.batching.BatchTracer))
```

Comment:
fyi (since you asked), back in jax v0.2.17, isinstance(my_jax_array, numpy.ndarray) would return True: https://github.com/google/jax/blob/jax-v0.2.17/jax/_src/numpy/lax_numpy.py#L93

so to support older versions of jax we had to add the check `not isinstance(a, onp.ndarray)`

Reply:
I see, that makes sense.